lib/jekyll/convertible.rb

```diff
@@ -61,6 +61,10 @@ def read_yaml(base, name, opts = {})
         Jekyll.logger.abort_with "Fatal:", "Invalid YAML front matter in #{File.join(base, name)}"
       end
 
+      unless valid_permalink?(self.data['permalink'])
```

Comment:
And please add a test to ensure this output is not written if the user did not specify a permalink in the front matter.